### DISCUSSION TITLE
Chat History

### DISCUSSION THREAD

**Discussion by hibaq86** [FlowiseAI/Flowise#1250] [Q&A]
When I embed the app into my website, will it remember the chat history of all users? Whenever I test out my full page HTML, it remembers my previous questions until I press reset.

**Comment by chungyau97** [comment]
Hi @hibaq86, yes it will store all messages on their local storage. so each users will have their own history

**Comment by hibaq86** [comment]
But I don’t have access to that conversation history. And does the app remember the conversation history based on the user’s IP address? Hibaq AbdiOn Nov 19, 2023, at 6:42 AM, Ong Chung Yau ***@***.***> wrote:﻿
Hi @hibaq86, yes it will store all messages on their local storage. so each users will have their own history

**Reply by chungyau97** [reply]
Hi @hibaq86 , in the current version you're able to retrieve all users messages shown below by going into your fflow's  settings and click view messages:
![image](https://github.com/FlowiseAI/Flowise/assets/33013947/1f67463a-8859-4c41-9e9f-ac9480035161)

Regarding on keeping tack on user's leads information like email, phone number and name will be in the future release.

**Comment by hibaq86** [comment]
I cannot find fflow settings! Hibaq AbdiMSW, RSW, BSW, OCGC(MKM), BA(MIT)On Nov 19, 2023, at 8:47 PM, Ong Chung Yau ***@***.***> wrote:﻿
Hi @hibaq86 , in the current version you're able to retrieve all users messages shown below by going into your fflow's  settings and click view messages:

Regarding on keeping tack on user's leads information like email, phone number and name will be in the future release.

**Reply by chungyau97** [reply]
Shown below is the settings of your flow:
![image](https://github.com/FlowiseAI/Flowise/assets/33013947/218a80f6-da72-481b-b850-f9e37dc163ee)

**Comment by hibaq86** [comment]
Can flowise the company see what my users are putting in the chat? Hibaq AbdiMSW, RSW, BSW, OCGC(MKM), BA(MIT)On Nov 20, 2023, at 2:24 AM, Ong Chung Yau ***@***.***> wrote:﻿
Shown below is the settings of your flow:

**Comment by hibaq86** [comment]
Also, I’m using vectara upload file, chat open AI, and conversational retrieval QA Chain. So I don’t see the message section in settings. I don’t care about seeing peoples messages, as long as I can make sure flowise doesn’t have access to my users messages. I have to make a privacy policy and need to know what type of information the flowise app is collecting of my users. Hibaq AbdiMSW, RSW, BSW, OCGC(MKM), BA(MIT)On Nov 20, 2023, at 2:24 AM, Ong Chung Yau ***@***.***> wrote:﻿
Shown below is the settings of your flow:

**Reply by chungyau97** [reply]
HI @hibaq86, flowise do not have access to your users chat messages. All your data is only available on the device you setup.
